src/icrc1_auction_api.mo

```diff
@@ -530,21 +530,36 @@ actor class Icrc1AuctionAPI(quoteLedger_ : ?Principal, adminPrincipal_ : ?Princi
 
   public shared ({ caller }) func btc_withdraw(args : { to : Text; amount : Nat; expected_fee : ?Nat }) : async BtcWithdrawResult {
     let ?ckbtcAssetId = getAssetId(CKBTC_LEDGER_PRINCIPAL) else throw Error.reject("BTC is not supported");
+    let handler = Vec.get(assets, ckbtcAssetId).handler;
 
     let (rollbackCredit, doneCallback) = switch (auction.deductCredit(caller, ckbtcAssetId, args.amount)) {
       case (#err _) return #Err(#InsufficientCredit({}));
       case (#ok(_, r, d)) (r, d);
     };
-    switch (
-      await* btcHandler.withdraw(args.to, args.amount, args.expected_fee)
+    let withdrawalResult = switch (
+      await* btcHandler.withdraw(
+        args.to,
+        args.amount,
+        Option.get<Nat>(args.expected_fee, handler.fee(#allowance)),
```

Comment:
Doesn't `handler.fee(#allowance)` include the surcharge?
What we need here is the ckBTC ledger fee, or?